File: app/actions/single-cp.ts
import * as csv from 'csv-writer';
import Stripe from 'stripe';

const stripe = new Stripe('pass_your_stripe_secret_key');

async function generateCoupon(): Promise<void> {
    const customId = "APPS80";
    const couponName = "AppSumo";
    try {
        const coupon = await stripe.coupons.create({
            name: couponName,
            duration: 'once',
            percent_off: 20,
            currency: 'usd',
            id: customId
        });
        
        const csvWriter = csv.createObjectCsvWriter({
            path: 'coupon.csv',
            header: [
                { id: 'Stripe ID', title: 'Stripe ID' },
                { id: 'Custom ID', title: 'Custom ID' },
                { id: 'Coupon Name', title: 'Coupon Name' }
            ]
        });
        
        await csvWriter.writeRecords([
            { 'Stripe ID': coupon.id, 'Custom ID': customId, 'Coupon Name': couponName }
        ]);
        
        console.log("Coupon generated and saved to coupon.csv.");
    } catch (error) {
        console.error("Error generating coupon:", error);
    }
}

generateCoupon();

File: single-cp.py
import csv
import stripe

stripe.api_key = 'pass_your_stripe_secret_key'

def generate_coupon():
    custom_id = "APPS30" 
    coupon_name = "AppSumo"  
    coupon = stripe.Coupon.create(
        name=coupon_name,
        duration='once',
        percent_off=20,  
        currency='usd',
        id=custom_id  
    )

    with open('coupon.csv', 'w', newline='') as csvfile:
        writer = csv.writer(csvfile)
        writer.writerow(["Stripe ID", "Custom ID", "Coupon Name"])
        writer.writerow([coupon.id, custom_id, coupon_name])

    print("Coupon generated and saved to coupon.csv.")

generate_coupon()